flexmeasures/conftest.py

```diff
@@ -231,34 +230,42 @@ def create_roles_users(db, test_accounts) -> dict[str, User]:
 
 
 @pytest.fixture(scope="module")
-def setup_markets(db) -> dict[str, Market]:
+def setup_markets(db) -> dict[str, Sensor]:
     return create_test_markets(db)
 
 
 @pytest.fixture(scope="function")
-def setup_markets_fresh_db(fresh_db) -> dict[str, Market]:
+def setup_markets_fresh_db(fresh_db) -> dict[str, Sensor]:
     return create_test_markets(fresh_db)
 
 
-def create_test_markets(db) -> dict[str, Market]:
+def create_test_markets(db) -> dict[str, Sensor]:
     """Create the epex_da market."""
 
-    day_ahead = MarketType(
+    day_ahead = GenericAssetType(
         name="day_ahead",
-        daily_seasonality=True,
-        weekly_seasonality=True,
-        yearly_seasonality=True,
     )
-    db.session.add(day_ahead)
-    epex_da = Market(
+    epex = GenericAsset(
+        name="epex",
+        generic_asset_type=day_ahead,
+    )
+    epex_da = Sensor(
         name="epex_da",
-        market_type_name="day_ahead",
+        generic_asset=epex,
         event_resolution=timedelta(hours=1),
         unit="EUR/MWh",
-        knowledge_horizon_fnc="x_days_ago_at_y_oclock",
-        knowledge_horizon_par={"x": 1, "y": 12, "z": "Europe/Paris"},
+        knowledge_horizon=(
+            x_days_ago_at_y_oclock,
+            {"x": 1, "y": 12, "z": "Europe/Paris"},
+        ),
+        attributes=dict(
+            daily_seasonality=True,
+            weekly_seasonality=True,
+            yearly_seasonality=True,
+        ),
     )
     db.session.add(epex_da)
+    db.session.flush()  # assign an id so it can be used to set a market_id attribute on an Asset or Sensor
```

Comment:
This comment sounds a bit like legacy - "market_id", "Asset"?
Maybe this line doesn't need to be added after all?

Reply:
It *does* sound legacy. The name "market_id" is still in use as the attribute name that schedulers look for as a fallback in case of a missing pricing sensor in the flex context. The flush is actually needed for tests to pass.

I meant "GenericAsset" here, which I'm now correcting, I hope that helps.

Reply:
I was [planning](https://github.com/FlexMeasures/flexmeasures/discussions/763#discussioncomment-6500979) to document which attributes are actually relevant, and this one would have to be added to the list.

Is it possible to replace the name `market_id` with `consumption_price_sensor`, for instance, so we get more consistency?

Reply:
Yes. Requires a db migration, and we should check plugins (maybe only flexmeasures-simulate, assuming E-mission is deprecated). I may also recommend a soft deprecation in internal code, by checking for the old attribute name as a fallback (for a few versions), to avoid problems with e.g. ongoing simulation projects. I'm guessing you want to move it into a nested structure under a "flex-context" attribute? There the parameter name should be dashed instead of having underscores.

Reply:
Well, we already have `consumption-price-sensor` in the flex context. I was suggesting to rename `market_id` (where used). This can be a follow-up ticket, of course.

Reply:
Alright, definitely make it a follow-up ticket. Hopefully that will clear up what you have in mind.

Reply:
#774